backend/src/v2/compiler/argocompiler/common.go

```diff
@@ -46,6 +56,80 @@ var commonEnvs = []k8score.EnvVar{{
 	},
 }}
 
+func GetMLPipelineServiceTLSEnabled() (bool, error) {
+	mlPipelineServiceTLSEnabledStr := os.Getenv(MLPipelineTLSEnabledEnvVar)
+	if mlPipelineServiceTLSEnabledStr == "" {
+		return DefaultMLPipelineTLSEnabled, nil
+	}
+	mlPipelineServiceTLSEnabled, err := strconv.ParseBool(os.Getenv(MLPipelineTLSEnabledEnvVar))
+	if err != nil {
+		return false, err
+	}
+	return mlPipelineServiceTLSEnabled, nil
+}
+
+// ConfigureCABundle adds CABundle environment variables and volume mounts
+// if CA Bundle env vars are specified.
+func ConfigureCABundle(tmpl *wfapi.Template) {
+	caBundleCertPath := os.Getenv(common.CaBundlePath)
+	caBundleDir := path.Dir(caBundleCertPath)
+	caBundleSecretName := os.Getenv(common.CaBundleSecretName)
+	if caBundleSecretName != "" {
+		var certDirectories = []string{
+			caBundleDir,
+			"/etc/ssl/certs",
+			"/etc/pki/tls/certs",
+		}
+		// Add to REQUESTS_CA_BUNDLE for python request library.
+		// As many python web based libraries utilize this, we add it here so the user
+		// does not have to manually include this in the user pipeline.
+		// Note: for packages like Boto3, even though it is documented to use AWS_CA_BUNDLE,
+		// we found the python boto3 client only works if we include REQUESTS_CA_BUNDLE.
+		// https://requests.readthedocs.io/en/latest/user/advanced/#ssl-cert-verification
+		// https://github.com/aws/aws-cli/issues/3425
+		tmpl.Container.Env = append(tmpl.Container.Env, k8score.EnvVar{
```

Comment:
The launcher handles this now so we can remove these environment variables on the compiler.